<a href="https://colab.research.google.com/github/sitthatkaja/DWDM21/blob/main/Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#ปัญหา : ต้องการพยากรณ์ว่าใช้ประเภทของระบบกำจัดของเสียประเภทใดมีการใช้มากที่สุด

#ใช้ข้อมูลทั้งหมด 3 ชุด 
- ข้อมูลระบบกำจัดของเสีย อ้างอิงจาก : https://data.go.th/
- ข้อมูลค่าเฉลี่ยประชากร อ้างอิงจาก : https://data.go.th/
- ข้อมูลรายได้ประชารกร อ้างอิงจาก : https://data.go.th/

ไฟล์ที่ใช้ทั้งหมด : https://drive.google.com/drive/folders/1nETPdrCX90tF_F-4eNZQIacxdAMNrqUQ?usp=sharing

In [ ]:
import pandas as pd #ทำงานกับข้อมูลลักษณะตาราง

In [ ]:
from google.colab import drive #เชื่อม Google drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os # os ไว้จัดการไฟล์และถาท ถาทคือที่อยู่ของไฟล์

In [ ]:
path = '/content/drive/My Drive/DWDM2021_DATA' #ชื่อ path ชี้ไปไฟล์ที่เราต้องการ
data_BB = pd.read_csv(os.path.join(path,'กำจัดของเสีย.csv'))
data_BB #ข้อมูลหลักชุดที่ 1

ที่  ...                                      แหล่งงบประมาณ
0     1  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
1     2  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
2     3  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
3     4  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
4     5  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
..  ...  ...                                                ...
89   90  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
90   91  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
91   92  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
92   93  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
93   94  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม

[94 rows x 12 columns]

In [ ]:
data_PP = pd.read_csv(os.path.join(path,'ค่าเฉลี่ยประชากร.csv'))
data_PP.dropna() #ข้อมูลรองชุดที่ 2

จังหวัด  15 - 39 ปี  40 - 59 ปี  ...  15 ปีขึ้นไป  60 ปีขึ้นไป  ผลเฉลี่ย
0   กรุงเทพมหานคร       12.30       10.87  ...        11.11         8.13     54.07
1     สมุทรปราการ       11.17        9.38  ...         9.88         7.29     48.06
2         นนทบุรี       12.12       10.87  ...        10.86         8.06     53.40
3        ปทุมธานี       11.33       10.20  ...        10.31         7.08     49.76
4          อยุธยา       11.11        9.04  ...         9.32         5.47     45.20
..            ...         ...         ...  ...          ...          ...       ...
72           ตรัง       10.16        8.31  ...         8.45         5.06     41.27
73         พัทลุง       10.83        8.67  ...         8.51         4.83     42.59
74        ปัตตานี        9.77        7.03  ...         7.70         3.67     36.85
75           ยะลา       10.69        8.05  ...         8.63         3.56     40.54
76       นราธิวาส        9.22        6.51  ...         7.23         2.97     34.05

[77 rows x 7 columns]

In [ ]:
data_PC = pd.read_csv(os.path.join(path,'รายได้เฉลี่ย.csv'))
data_PC.dropna() #ข้อมูลรองชุดที่ 3

จังหวัด รหัสจังหวัด  ...        2561     รวมเฉลี่ย
0   กรุงเทพมหานคร       th-bm  ...  1051118.51  5.383503e+06
1       กาญจนบุรี       th-kn  ...    27138.35  1.295467e+05
2        จันทบุรี       th-ct  ...     8469.45  4.042424e+04
3      ฉะเชิงเทรา       th-cc  ...     5007.28  2.897594e+04
4          ชลบุรี       th-cb  ...   272435.77  1.144431e+06
..            ...         ...  ...         ...           ...
72          ระนอง       th-rn  ...     4706.13  2.589797e+04
73          สงขลา       th-sg  ...    68097.42  3.339657e+05
74           สตูล       th-sa  ...     9101.47  4.895520e+04
75   สุราษฎร์ธานี       th-st  ...   104503.71  4.825130e+05
76         บึงกาฬ       th-bk  ...     1047.13  4.415360e+03

[77 rows x 10 columns]

#ดูว่าข้อมูลมี data missing

In [ ]:
data_BB.isnull() #ดูว่าข้อมูลนั้นมี Missing รึป่าว

ที่    ภาค  ...   ปริมาณน้ำเสียที่รองรับได้ (ลบ.ม/วัน)   แหล่งงบประมาณ
0   False  False  ...                                   False          False
1   False  False  ...                                   False          False
2   False  False  ...                                   False          False
3   False  False  ...                                   False          False
4   False  False  ...                                   False          False
..    ...    ...  ...                                     ...            ...
89  False  False  ...                                   False          False
90  False  False  ...                                   False          False
91  False  False  ...                                   False          False
92  False  False  ...                                   False          False
93  False  False  ...                                   False          False

[94 rows x 12 columns]

In [ ]:
data_PP.isnull() #ดูว่าข้อมูลนั้นมี Missing รึป่าว

จังหวัด  15 - 39 ปี  40 - 59 ปี  ...  15 ปีขึ้นไป  60 ปีขึ้นไป  ผลเฉลี่ย
0     False       False       False  ...        False        False     False
1     False       False       False  ...        False        False     False
2     False       False       False  ...        False        False     False
3     False       False       False  ...        False        False     False
4     False       False       False  ...        False        False     False
..      ...         ...         ...  ...          ...          ...       ...
72    False       False       False  ...        False        False     False
73    False       False       False  ...        False        False     False
74    False       False       False  ...        False        False     False
75    False       False       False  ...        False        False     False
76    False       False       False  ...        False        False     False

[77 rows x 7 columns]

In [ ]:
data_PC.isnull() #ดูว่าข้อมูลนั้นมี Missing รึป่าว

จังหวัด  รหัสจังหวัด   2555   2556  ...   2559   2560   2561  รวมเฉลี่ย
0     False        False  False  False  ...  False  False  False      False
1     False        False  False  False  ...  False  False  False      False
2     False        False  False  False  ...  False  False  False      False
3     False        False  False  False  ...  False  False  False      False
4     False        False  False  False  ...  False  False  False      False
..      ...          ...    ...    ...  ...    ...    ...    ...        ...
72    False        False  False  False  ...  False  False  False      False
73    False        False  False  False  ...  False  False  False      False
74    False        False  False  False  ...  False  False  False      False
75    False        False  False  False  ...  False  False  False      False
76    False        False  False  False  ...  False  False  False      False

[77 rows x 10 columns]

#สรุปข้อมูลเป็นรายคอลลัมม์ว่ามี missing

In [ ]:
data_BB.isnull().any() #any สรุปข้อมูลเป็นรายคอลลัมม์ว่ามี missing ไหม

ที่                                       False
ภาค                                       False
จังหวัด                                   False
หน่วยงานที่รับผิดชอบ                      False
ที่ตั้ง                                   False
ชื่อย่อของระบบ                            False
ประเภทของระบบ                             False
พ.ศ. ที่ได้รับงบประมาณ                    False
พ.ศ. ที่เปิดใช้งาน                         True
สถานภาพปัจจุบัน                           False
 ปริมาณน้ำเสียที่รองรับได้ (ลบ.ม/วัน)      True
แหล่งงบประมาณ                             False
dtype: bool

In [ ]:
data_PP.isnull().any() #any สรุปข้อมูลเป็นรายคอลลัมม์ว่ามี missing ไหม

จังหวัด        False
15 - 39 ปี     False
40 - 59 ปี     False
15 - 59 ปี     False
15 ปีขึ้นไป    False
60 ปีขึ้นไป    False
ผลเฉลี่ย       False
dtype: bool

In [ ]:
data_PC.isnull().any() #any สรุปข้อมูลเป็นรายคอลลัมม์ว่ามี missing ไหม

จังหวัด        False
รหัสจังหวัด    False
2555           False
2556           False
2557           False
2558           False
2559           False
2560           False
2561           False
รวมเฉลี่ย      False
dtype: bool

จากการตรวจสอบค่า missing value ของทั้ง 3 ชุดข้อมูลพบว่ามีค่าเป็น Ture อยู่ 3 colum ในข้อมูลชุดที่ 1 คือ พ.ศ. ที่เปิดใช้งาน และ ปริมาณน้ำเสียที่รองรับได้ (ลบ.ม/วัน)

#Drop missing value
เป็นการลบข้อมูลที่มีค่า missing ออกไปจากตาราง

In [ ]:
data_BB = data_BB.dropna()
data_BB

ที่  ...                                      แหล่งงบประมาณ
0     1  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
1     2  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
3     4  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
4     5  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
5     6  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
..  ...  ...                                                ...
88   89  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
89   90  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
90   91  ...  กระทรวงมหาดไทย กระทรวงทรัพยากรธรรมชาติและสิ่งแ...
91   92  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม
93   94  ...              กระทรวงทรัพยากรธรรมชาติและสิ่งแวดล้อม

[85 rows x 12 columns]

#ตรวสจสอบเมื่อลบออกไปแล้วตรวจสอบว่ายังไม่ข้อมูลที่หายไปหรือไม่

In [ ]:
data_BB.isnull().any()

ที่                                       False
ภาค                                       False
จังหวัด                                   False
หน่วยงานที่รับผิดชอบ                      False
ที่ตั้ง                                   False
ชื่อย่อของระบบ                            False
ประเภทของระบบ                             False
พ.ศ. ที่ได้รับงบประมาณ                    False
พ.ศ. ที่เปิดใช้งาน                        False
สถานภาพปัจจุบัน                           False
 ปริมาณน้ำเสียที่รองรับได้ (ลบ.ม/วัน)     False
แหล่งงบประมาณ                             False
dtype: bool

#Percent of missing data from dropna 
เป็นการตรวจสอบว่าเมื่อลบข้อมูลทิ้งไปแล้วเราสูญเสียข้อมูลไปกี่เปอร์เซ็นต์

In [ ]:
before = 94
after = 85
print(f'before = {before} \nafter = {after}\nwhich is {100*((before-after)/before)}')

before = 94 
after = 85
which is 9.574468085106384


จะเห็นได้ว่าค่าของข้อมูลที่สูญหายไปคิดเป็น 9.574468085106384 %

#รวมตาราง

In [ ]:
table = pd.merge(data_BB,data_PP,on='จังหวัด') #รวมตารางข้อมูลชุดที่ 1 และ 2

In [ ]:
table

ที่                 ภาค      จังหวัด  ... 15 ปีขึ้นไป 60 ปีขึ้นไป ผลเฉลี่ย
0     1               เหนือ     เชียงราย  ...        7.24        3.90    36.36
1     2               เหนือ     เชียงราย  ...        7.24        3.90    36.36
2     4               เหนือ    เชียงใหม่  ...        7.96        4.49    39.06
3     5               เหนือ         น่าน  ...        7.77        4.40    39.96
4     6               เหนือ        พะเยา  ...        7.84        4.45    40.28
..  ...                 ...          ...  ...         ...         ...      ...
73   89  ตะวันออกเฉียงเหนือ     อุดรธานี  ...        8.07        4.89    39.80
74   90  ตะวันออกเฉียงเหนือ     อุดรธานี  ...        8.07        4.89    39.80
75   91  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...        7.33        4.47    36.25
76   92  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...        7.33        4.47    36.25
77   94  ตะวันออกเฉียงเหนือ   อำนาจเจริญ  ...        7.70        4.70    38.66

[78 rows x 18 columns]

In [ ]:
table_3 = pd.merge(table,data_PC,on='จังหวัด') #รวมตารางข้อมูลชุดที่ 1 และ 2 และ 3

In [ ]:
table_3

ที่                 ภาค      จังหวัด  ...      2560       2561  รวมเฉลี่ย
0     1               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
1     2               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
2     4               เหนือ    เชียงใหม่  ...  99070.42  108012.95  557866.57
3     5               เหนือ         น่าน  ...   2462.93    2666.09   13634.01
4     6               เหนือ        พะเยา  ...   1286.52    1380.38    7862.51
..  ...                 ...          ...  ...       ...        ...        ...
69   89  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
70   90  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
71   91  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
72   92  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
73   94  ตะวันออกเฉียงเหนือ   อำนาจเจริญ  ...    429.93     449.62    2585.07

[74 rows x 27 columns]

In [ ]:
table_3.isnull().any()

ที่                                       False
ภาค                                       False
จังหวัด                                   False
หน่วยงานที่รับผิดชอบ                      False
ที่ตั้ง                                   False
ชื่อย่อของระบบ                            False
ประเภทของระบบ                             False
พ.ศ. ที่ได้รับงบประมาณ                    False
พ.ศ. ที่เปิดใช้งาน                        False
สถานภาพปัจจุบัน                           False
 ปริมาณน้ำเสียที่รองรับได้ (ลบ.ม/วัน)     False
แหล่งงบประมาณ                             False
15 - 39 ปี                                False
40 - 59 ปี                                False
15 - 59 ปี                                False
15 ปีขึ้นไป                               False
60 ปีขึ้นไป                               False
ผลเฉลี่ย                                  False
รหัสจังหวัด                               False
2555                                      False
2556                                    

In [ ]:
table_3.shape #ตาราง table_3 มี 77 แถว 33 colum

(74, 27)

In [ ]:
table_3.mean() #ตวจสอบค่าเฉลี่ยของแต่ละบริษัท

ที่                           43.554054
พ.ศ. ที่ได้รับงบประมาณ      2547.472973
พ.ศ. ที่เปิดใช้งาน          2549.783784
15 - 39 ปี                    10.619324
40 - 59 ปี                     7.901081
15 - 59 ปี                     9.314459
15 ปีขึ้นไป                    8.295541
60 ปีขึ้นไป                    4.948514
ผลเฉลี่ย                      41.078919
2555                       21061.964865
2556                       24107.887027
2557                       24559.745000
2558                       28074.892297
2559                       34835.735946
2560                       41877.167838
2561                       47024.057838
รวมเฉลี่ย                 221541.450811
dtype: float64

#ตรวจสอบดูการลบข้อมูลที่ซ้ำกัน

In [ ]:
table_3_un = table_3.drop_duplicates()
table_3_un

ที่                 ภาค      จังหวัด  ...      2560       2561  รวมเฉลี่ย
0     1               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
1     2               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
2     4               เหนือ    เชียงใหม่  ...  99070.42  108012.95  557866.57
3     5               เหนือ         น่าน  ...   2462.93    2666.09   13634.01
4     6               เหนือ        พะเยา  ...   1286.52    1380.38    7862.51
..  ...                 ...          ...  ...       ...        ...        ...
69   89  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
70   90  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
71   91  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
72   92  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
73   94  ตะวันออกเฉียงเหนือ   อำนาจเจริญ  ...    429.93     449.62    2585.07

[74 rows x 27 columns]

จากการตรวจสอบข้อมูลที่ซ้ำ ของ table พบว่าไม่มีค่าข้อมูลที่ซ้ำกัน

#Data Mining

#สร้างตารางที่ใช้ในการดู challenge

In [ ]:
data = table_3[['จังหวัด','ประเภทของระบบ','ผลเฉลี่ย']]
data

จังหวัด       ประเภทของระบบ  ผลเฉลี่ย
0      เชียงราย    ระบบสระเติมอากาศ     36.36
1      เชียงราย    ระบบบึงประดิษฐ์      36.36
2     เชียงใหม่    ระบบสระเติมอากาศ     39.06
3          น่าน   ระบบบ่อปรับเสถียร     39.96
4         พะเยา   ระบบบ่อปรับเสถียร     40.28
..          ...                 ...       ...
69     อุดรธานี   ระบบบ่อปรับเสถียร     39.80
70     อุดรธานี  ระบบรวบรวมน้ำเสีย      39.80
71  อุบลราชธานี    ระบบคลองวนเวียน      36.25
72  อุบลราชธานี   ระบบบ่อปรับเสถียร     36.25
73   อำนาจเจริญ   ระบบบ่อปรับเสถียร     38.66

[74 rows x 3 columns]

In [ ]:
data.shape #ข้อมูลนี้มี 77 แถว 4 คอลัม

(74, 3)

In [ ]:
data.isnull().any() #ตรวจสอบว่าตารางมีค่าว่างหรือไม่ จะเห็นว่าไม่คอลัมน์ที่มีค่าว่าง

จังหวัด          False
ประเภทของระบบ    False
ผลเฉลี่ย         False
dtype: bool

#สร้างตารางที่ค่าเฉลี่ยนประชากรมากกว่า 40

In [ ]:
data = table_3[table_3['ผลเฉลี่ย']>=40]
data

ที่                 ภาค        จังหวัด  ...       2560       2561   รวมเฉลี่ย
4     6               เหนือ          พะเยา  ...    1286.52    1380.38     7862.51
5     7               เหนือ          ลำปาง  ...    3773.95    4202.99    20488.05
6     8               เหนือ          ลำพูน  ...    1476.08    1593.53     8878.97
7     9               เหนือ      อุตรดิตถ์  ...    2071.57    2206.06    11315.41
8    10                 ใต้         กระบี่  ...  105029.06  118676.13   570696.47
9    11                 ใต้         กระบี่  ...  105029.06  118676.13   570696.47
10   12                 ใต้           ตรัง  ...    9028.01    9916.29    54376.28
11   13                 ใต้  นครศรีธรรมราช  ...   15609.04   17136.46    91429.56
12   14                 ใต้  นครศรีธรรมราช  ...   15609.04   17136.46    91429.56
13   15                 ใต้         ภูเก็ต  ...  423012.85  477324.35  2317913.01
14   16                 ใต้         ภูเก็ต  ...  423012.85  477324.35  2317913.01
15   17                 ใต้         ภูเก็ต  ...  423012.85  477324.35  2317913.01
16   18                 ใต้           ยะลา  ...    3145.57    3443.17    18787.07
17   19                 ใต้           ยะลา  ...    3145.57    3443.17    18787.07
18   20                 ใต้          สงขลา  ...   59831.77   68097.42   333965.68
19   21                 ใต้          สงขลา  ...   59831.77   68097.42   333965.68
22   25                กลาง      นครสวรรค์  ...    3641.46    3899.68    20104.86
23   26                กลาง      นครสวรรค์  ...    3641.46    3899.68    20104.86
24   27                กลาง      นครสวรรค์  ...    3641.46    3899.68    20104.86
25   28                กลาง         พิจิตร  ...    1421.33    1486.74     7068.58
26   29                กลาง         พิจิตร  ...    1421.33    1486.74     7068.58
27   30                กลาง       พิษณุโลก  ...    7904.34    8477.09    48885.70
28   31                กลาง       พิษณุโลก  ...    7904.34    8477.09    48885.70
29   32                กลาง       พิษณุโลก  ...    7904.34    8477.09    48885.70
32   35                กลาง         ชัยนาท  ...    1236.34    1314.22     6377.28
33   36                กลาง         ชัยนาท  ...    1236.34    1314.22     6377.28
35   38                กลาง        นนทบุรี  ...    3963.09    4368.99    22036.32
36   39                กลาง        นนทบุรี  ...    3963.09    4368.99    22036.32
38   43                กลาง         ลพบุรี  ...    5304.47    5816.17    32165.77
39   44                กลาง      สมุทรสาคร  ...    2478.32    2648.22    14638.43
40   45                กลาง      สมุทรสาคร  ...    2478.32    2648.22    14638.43
42   47                กลาง        อ่างทอง  ...     920.84     986.35     5322.88
45   54             ตะวันตก       เพชรบุรี  ...   29413.34   31652.99   155268.69
46   55             ตะวันตก       เพชรบุรี  ...   29413.34   31652.99   155268.69
47   56             ตะวันตก       เพชรบุรี  ...   29413.34   31652.99   155268.69
48   57             ตะวันตก        ราชบุรี  ...    3781.35    4197.69    17868.95
49   58             ตะวันตก        ราชบุรี  ...    3781.35    4197.69    17868.95
50   59             ตะวันตก        ราชบุรี  ...    3781.35    4197.69    17868.95
51   60            ตะวันออก       จันทบุรี  ...    7628.47    8469.45    40424.24
52   61            ตะวันออก         ชลบุรี  ...  240610.45  272435.77  1144431.15
53   63            ตะวันออก         ชลบุรี  ...  240610.45  272435.77  1144431.15
54   64            ตะวันออก         ชลบุรี  ...  240610.45  272435.77  1144431.15
55   65            ตะวันออก         ชลบุรี  ...  240610.45  272435.77  1144431.15
56   67            ตะวันออก     ฉะเชิงเทรา  ...    4655.55    5007.28    28975.94
57   68            ตะวันออก     ฉะเชิงเทรา  ...    4655.55    5007.28    28975.94
58   69            ตะวันออก          ระยอง  ...   33811.36   37023.96   198263.24
60   71  ตะวันออกเฉียงเหนือ        ขอนแก่น  ...   15608.33   17017.80    84595.53
61   72  ตะวันออกเฉียงเหนือ        ขอนแก่น  ...   15608.33   17017.8

#แปลงให้ข้อมูลอยู่ในรูปแบบเป็น transaction

In [ ]:
table = data.groupby('จังหวัด')['ประเภทของระบบ'].apply(list)
table

จังหวัด
กระบี่               [ระบบตะกอนเร่ง+บึงประดิษฐ์ , ระบบกลุ่มอาคาร ]
ขอนแก่น                       [ระบบสระเติมอากาศ, ระบบบึงประดิษฐ์ ]
จันทบุรี                                       [ระบบบ่อปรับเสถียร]
ฉะเชิงเทรา                   [ระบบคลองวนเวียน , ระบบบ่อปรับเสถียร]
ชลบุรี           [ระบบบ่อปรับเสถียร, ระบบสระเติมอากาศ, ระบบบึงป...
ชัยนาท                        [ระบบสระเติมอากาศ, ระบบบึงประดิษฐ์ ]
ตรัง                                            [ระบบสระเติมอากาศ]
นครศรีธรรมราช              [ระบบบ่อปรับเสถียร, ระบบจานหมุนชีวภาพ ]
นครสวรรค์        [ระบบตะกอนเร่งแบบชีวมวล , ระบบบ่อปรับเสถียร, ร...
นนทบุรี                           [ระบบตะกอนเร่ง , ระบบตะกอนเร่ง ]
พะเยา                                          [ระบบบ่อปรับเสถียร]
พิจิตร                       [ระบบสระเติมอากาศ, ระบบบ่อปรับเสถียร]
พิษณุโลก         [ระบบบ่อปรับเสถียร, ระบบกลุ่มอาคาร , ระบบกลุ่ม...
ภูเก็ต           [ระบบคลองวนเวียน , ระบบคลองวนเวียน , ระบบคลองว...
มหาสารคาม                                      [ระบบบ่

#ขั้นตอนติดตั้ง apyori

In [ ]:
!pip install apyori

  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5974 sha256=3ddc68bdfa5873df874a52b5e42f0cf631a12b8c13bd550cf4977ad252e41b20
  Stored in directory: /root/.cache/pip/wheels/cb/f6/e1/57973c631d27efd1a2f375bd6a83b2a616c4021f24aab84080
Successfully built apyori


#เป็นการเรียกใช้ฟังก์ชัน apriori

In [ ]:
from apyori import apriori

#ผลลัพธ์ที่ได้จากการทำ Associantion ซึ่งรายละเอียดจะอยู่ที่ summary

In [ ]:
test_1 = list(apriori(table, min_support = 0.039))
test_1

[RelationRecord(items=frozenset({'ระบบกลุ่มอาคาร '}), support=0.15384615384615385, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบกลุ่มอาคาร '}), confidence=0.15384615384615385, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบคลองวนเวียน '}), support=0.15384615384615385, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบคลองวนเวียน '}), confidence=0.15384615384615385, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบบึงประดิษฐ์ '}), support=0.15384615384615385, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบบึงประดิษฐ์ '}), confidence=0.15384615384615385, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบบ่อปรับเสถียร'}), support=0.5384615384615384, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบบ่อปรับเสถียร'}), confidence=0.5384615384615384, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบสระเติมอากาศ'}), support=0.34615

#สร้างตารางที่ค่าเฉลี่ยนประชากรน้อยกว่า 40

In [ ]:
dataP = table_3[table_3['ผลเฉลี่ย']<=40]
dataP

ที่                 ภาค      จังหวัด  ...      2560       2561  รวมเฉลี่ย
0     1               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
1     2               เหนือ     เชียงราย  ...  26053.86   28548.43  160035.50
2     4               เหนือ    เชียงใหม่  ...  99070.42  108012.95  557866.57
3     5               เหนือ         น่าน  ...   2462.93    2666.09   13634.01
20   23                กลาง    กำแพงเพชร  ...   1529.10    1616.48    8696.77
21   24                กลาง    กำแพงเพชร  ...   1529.10    1616.48    8696.77
22   25                กลาง    นครสวรรค์  ...   3641.46    3899.68   20104.86
23   26                กลาง    นครสวรรค์  ...   3641.46    3899.68   20104.86
24   27                กลาง    นครสวรรค์  ...   3641.46    3899.68   20104.86
30   33                กลาง      สุโขทัย  ...   3510.33    3717.76   20711.74
31   34                กลาง    อุทัยธานี  ...   1310.51    1391.68    6503.98
34   37                กลาง       นครปฐม  ...   5549.04    6014.59   30742.27
37   42                กลาง    เพชรบูรณ์  ...   7101.23    7556.40   41304.84
41   46                กลาง   สุพรรณบุรี  ...   5207.53    5607.03   26117.01
43   48             ตะวันตก          ตาก  ...   6604.26    7131.48   39366.10
44   49             ตะวันตก          ตาก  ...   6604.26    7131.48   39366.10
59   70  ตะวันออกเฉียงเหนือ    กาฬสินธุ์  ...   1087.87    1165.77    5937.37
62   75  ตะวันออกเฉียงเหนือ      ชัยภูมิ  ...   2043.97    2202.45   11077.54
63   76  ตะวันออกเฉียงเหนือ       นครพนม  ...   1908.82    2106.51   10996.05
65   85  ตะวันออกเฉียงเหนือ        ยโสธร  ...    776.33     820.05    4507.88
66   86  ตะวันออกเฉียงเหนือ       สกลนคร  ...   2137.13    2333.98   12634.01
67   87  ตะวันออกเฉียงเหนือ     สุรินทร์  ...   2675.90    3026.05   16185.84
68   88  ตะวันออกเฉียงเหนือ     สุรินทร์  ...   2675.90    3026.05   16185.84
69   89  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
70   90  ตะวันออกเฉียงเหนือ     อุดรธานี  ...   9892.51   10786.68   58165.80
71   91  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
72   92  ตะวันออกเฉียงเหนือ  อุบลราชธานี  ...   7375.12    7990.17   42377.14
73   94  ตะวันออกเฉียงเหนือ   อำนาจเจริญ  ...    429.93     449.62    2585.07

[28 rows x 27 columns]

#แปลงให้ข้อมูลอยู่ในรูปแบบเป็น transaction

In [ ]:
table2 = dataP.groupby('จังหวัด')['ประเภทของระบบ'].apply(list)
table2

จังหวัด
กาฬสินธุ์                                     [ระบบสระเติมอากาศ]
กำแพงเพชร                 [ระบบบ่อปรับเสถียร, ระบบบ่อปรับเสถียร]
ชัยภูมิ                                       [ระบบสระเติมอากาศ]
ตาก                       [ระบบบ่อปรับเสถียร, ระบบบ่อปรับเสถียร]
นครปฐม                                        [ระบบสระเติมอากาศ]
นครพนม                                       [ระบบบ่อปรับเสถียร]
นครสวรรค์      [ระบบตะกอนเร่งแบบชีวมวล , ระบบบ่อปรับเสถียร, ร...
น่าน                                         [ระบบบ่อปรับเสถียร]
ยโสธร                                        [ระบบบ่อปรับเสถียร]
สกลนคร                          [ระบบบ่อปรับเสถียร+บึงประดิษฐ์ ]
สุพรรณบุรี                                   [ระบบบ่อปรับเสถียร]
สุรินทร์                   [ระบบบ่อปรับเสถียร, ระบบบึงประดิษฐ์ ]
สุโขทัย                                      [ระบบบ่อปรับเสถียร]
อำนาจเจริญ                                   [ระบบบ่อปรับเสถียร]
อุดรธานี                 [ระบบบ่อปรับเสถียร, ระบบรวบรวมน้ำเสีย ]
อุทัยธานี        

#ผลลัพธ์ที่ได้จากการทำ Associantion ซึ่งรายละเอียดจะอยู่ที่ summary

In [ ]:
test_2 = list(apriori(table2, min_support = 0.06 ))
test_2

[RelationRecord(items=frozenset({'ระบบบึงประดิษฐ์ '}), support=0.15, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบบึงประดิษฐ์ '}), confidence=0.15, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบบ่อปรับเสถียร'}), support=0.65, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบบ่อปรับเสถียร'}), confidence=0.65, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบสระเติมอากาศ'}), support=0.25, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบสระเติมอากาศ'}), confidence=0.25, lift=1.0)]),
 RelationRecord(items=frozenset({'ระบบบ่อปรับเสถียร', 'ระบบบึงประดิษฐ์ '}), support=0.1, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ระบบบ่อปรับเสถียร', 'ระบบบึงประดิษฐ์ '}), confidence=0.1, lift=1.0), OrderedStatistic(items_base=frozenset({'ระบบบึงประดิษฐ์ '}), items_add=frozenset({'ระบบบ่อปรับเสถียร'}), confidence=0.6666666666666667, lift=1.02564

#SUMMARY

In [ ]:
summary_1 = pd.read_csv(os.path.join(path,'มากกว่า 40.csv'))
summary_1

SUMMARY > 40 Unnamed: 1
0                                   ระบบ     ร้อยละ
1                         ระบบกลุ่มอาคาร     15.38%
2                        ระบบคลองวนเวียน     15.38%
3                        ระบบบึงประดิษฐ์     15.38%
4                      ระบบบ่อปรับเสถียร     53.84%
5                       ระบบสระเติมอากาศ     34.61%
6   ระบบบ่อปรับเสถียร', 'ระบบคลองวนเวียน      7.69%
7  ระบบบึงประดิษฐ์ ', 'ระบบบ่อปรับเสถียร      7.69%
8    ระบบสระเติมอากาศ', 'ระบบบึงประดิษฐ์     11.53%
9  ระบบสระเติมอากาศ', 'ระบบบ่อปรับเสถียร     15.38%

จะเห็นได้ว่าค่าเฉลี่ยนของประชากรมากกว่า 40 มักจะนิยมใช้ระบบกำจัดของเสียแบบบ่อปรับเสถียรมากที่สุด

In [ ]:
summary_2 = pd.read_csv(os.path.join(path,'น้อยกว่า 40.csv'))
summary_2

SUMMARY <  40 Unnamed: 1
0                                   ระบบ     ร้อยละ
1                        ระบบบึงประดิษฐ์        15%
2                      ระบบบ่อปรับเสถียร        65%
3                       ระบบสระเติมอากาศ        25%
4  ระบบบึงประดิษฐ์ ', 'ระบบบ่อปรับเสถียร        10%

จะเห็นได้ว่าค่าเฉลี่ยนของประชากรน้อยกว่า 40 มักจะนิยมใช้ระบบกำจัดของเสียแบบบ่อปรับเสถียรมากที่สุดเช่นเดียวกัน